graylog2-web-interface/src/components/indices/rotation/TimeBasedSizeOptimizingStrategyConfiguration.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+
+import useMaxIndexRotationLimit from 'hooks/useMaxIndexRotationLimit';
+import RangeInput from 'components/common/RangeInput';
+
+export type TimeBasedSizeOptimizingStrategyConfig = {
+  index_lifetime_max: string,
+  index_lifetime_min: string,
+  type: string,
+}
+
+type Props = {
+  config: TimeBasedSizeOptimizingStrategyConfig,
+  updateConfig: (config: Omit<TimeBasedSizeOptimizingStrategyConfig, 'type'>) => void,
+}
+
+export const durationToRoundedDays = (duration: string) => {
+  return Math.round(moment.duration(duration).asDays());
+};
+
+const getRangeInDays = (indexLifeTimeMin = 'PT1H', IndexLifeTimeMax = 'P4D') => {
+  return [durationToRoundedDays(indexLifeTimeMin), durationToRoundedDays(IndexLifeTimeMax)];
+};
+
+const YEARS_DAYS = 365;
+
+const getInitialMaxRange = (maxRotationPeriod: number, maxLifetime: number) => {
+  if (maxRotationPeriod) {
+    return maxRotationPeriod;
+  }
+
+  return maxLifetime > YEARS_DAYS ? maxLifetime + YEARS_DAYS : YEARS_DAYS;
+};
+
+const TimeBasedSizeOptimizingStrategyConfiguration = ({ config: { index_lifetime_max, index_lifetime_min }, updateConfig }: Props) => {
+  const [indexLifetimeRange, setIndexLifetimeRange] = useState(getRangeInDays(index_lifetime_min, index_lifetime_max));
+  const maxRotationPeriod = useMaxIndexRotationLimit();
+  const [maxRange, setMaxRange] = useState(getInitialMaxRange(durationToRoundedDays(maxRotationPeriod), indexLifetimeRange[1]));
+
+  const _isValidRange = useCallback((range: Array<number>) => {
```

Comment:
Can we get rid of the `_`?
```suggestion
  const isValidRange = useCallback((range: Array<number>) => {
```